File: ls.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"os"
	"path"
)

type ListDirectory struct {
	directory string
}

func isDirectory(filename string) bool {
	file, err := os.Open(filename)
	if err != nil {
		return false
	}
	fileInfo, err := file.Stat()
	if err != nil {
		return false
	}
	return fileInfo.Mode().IsDir()
}

func (ls ListDirectory) executeLsCommand() {
	if _, err := os.Stat(ls.directory); os.IsNotExist(err) {
		fmt.Println(string("\033[31m"), "Directory does not exist")
		os.Exit(1)
	}
	content, err := ioutil.ReadDir(ls.directory)
	if err != nil {
		log.Fatal(err)
	}
	for _, file := range content {
		var name string = file.Name()
		var size int = int(file.Size())
		var isDir bool = isDirectory(path.Join(ls.directory, name))
		var color string = ""
		if isDir {
			color = "\033[34m"
		} else {
			color = "\033[33m"
		}

		nameString := string(color) + name
		fmt.Println(nameString, " -> ", size)
	}
}

func main() {
	var arguments []string = os.Args[1:]
	var directory string = ""
	if len(arguments) > 0 {
		directory = arguments[0]
	} else {
		directory, _ = os.Getwd()
	}
	var ls ListDirectory = ListDirectory{directory: directory}
	ls.executeLsCommand()
}

File: ls.ts
import { existsSync, readdir, statSync } from "fs";
import { blueBright, Chalk, cyanBright, redBright, yellowBright } from 'chalk'
import { cwd } from "process";
import { join } from "path";

class ListDirectories {
    private readonly directory:string;

    constructor(directory:string) {
        this.directory = directory;
        this.executeLsCommand(this.directory)
    }

    private executeLsCommand(directory:string):void {
        if(!this.directoryExists(directory)){
            console.log(redBright(`ERROR: ${directory} is not a directory`))
            process.exit()
        }

        readdir(directory, (err:any, content:Array<string>):void => {
            if(err) {
                console.log(redBright(`ERROR: ${err.msg}`))
                process.exit()
            }
            content.forEach((filename:string):void => {
                const size:number = statSync(filename).size
                const isDir:boolean = this.directoryExists(join(directory, filename))
                const color:Chalk = isDir ? blueBright : yellowBright
                const sizeString:string = `size:${size} bytes`
                console.log(`${color(filename)} -> [${cyanBright(sizeString)}]`)

            })
        })
    }

    private directoryExists(directory:string):boolean {
        try {
            const exists:boolean = existsSync(directory)
            const isDir:boolean = statSync(directory).isDirectory()
            return exists &&  isDir  
        } catch(exception) {
            return false
        }
    }
}

let args = process.argv;
args = args.slice(2)

console.log(args[1])

const ls = new ListDirectories(args[0] || cwd())